framework/auth/core.py

```diff
@@ -177,7 +177,15 @@ def private_link(self):
 
     @classmethod
     def from_kwargs(cls, request_args, kwargs):
-        user = request_args.get('user') or kwargs.get('user') or _get_current_user()
+        from osf.models import ApiOAuth2PersonalToken
```

Comment:
Drive-by: Do both of these need to be imported here?

Reply:
Yea sadly, importing at top of `core.py` gives a circular import error:
```
File "/Users/erin/code/osf_stuff/osf.io/addons/osfstorage/apps.py", line 1, in <module>
    from addons.base.apps import BaseAddonAppConfig
  File "/Users/erin/code/osf_stuff/osf.io/addons/base/apps.py", line 11, in <module>
    from website.util import rubeus
  File "/Users/erin/code/osf_stuff/osf.io/website/util/rubeus.py", line 11, in <module>
    from framework.auth.decorators import Auth
  File "/Users/erin/code/osf_stuff/osf.io/framework/auth/__init__.py", line 6, in <module>
    from framework.auth.core import Auth
  File "/Users/erin/code/osf_stuff/osf.io/framework/auth/core.py", line 13, in <module>
    from framework.auth.cas import parse_auth_header
  File "/Users/erin/code/osf_stuff/osf.io/framework/auth/cas.py", line 11, in <module>
    from framework.auth import authenticate, external_first_login_authenticate
ImportError: cannot import name authenticate
```